introduce composer to sugarcrm 6.5

jakubpas opened the pull request:
- introduces composer
- introduces psr-4 autoloading for Sugar stuff and proper namespacing
- doesn't break backward compatibility
- reduces unnecessary "requre" directives and allow easier refactoring
- allows to update in future old vendors software like Zend libraries, recaptcha, tcpdf and so on & move it to the vendor dir.

butschster commented:
https://github.com/butschster/sugarcrm_dev

jakubpas commented:
?

butschster commented:
Are you using SugarCRM CE ?

jakubpas commented:
sugarcrm 6.5

On Mon, Jan 18, 2016 at 6:47 PM, butschster <notifications@github.com>
wrote:

> Are you using SugarCRM CE ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sugarcrm/sugarcrm_dev/pull/233#issuecomment-172603576>
> .
>

butschster commented:
I started to improve SugarCRM 6.5 CE. And i moved at first standart php libraries (phpmailer, tcpdf, zf, tcpdf, e.t.c) to composer https://github.com/butschster/sugarcrm_dev/blob/master/composer.json.

I want to union SugarCRM with Laravel5.
I added console support in SugarCRM, Monolog, generator _ide_helper.php https://github.com/butschster/sugarcrm_dev/blob/master/_ide_helper.php with phpDoc for all beans.

Perhaps you might be interested.

jakubpas commented:
Sound interesting. I will take a look. I made some improvements to Sugar
myself. Perhaps I can contribute. One question - what the CE stnads for?

On Mon, Jan 18, 2016 at 7:04 PM, butschster <notifications@github.com>
wrote:

> I started to improve SugarCRM 6.5 CE. And i moved at first standart php
> libraries (phpmailer, tcpdf, zf, tcpdf, e.t.c) to composer
> https://github.com/butschster/sugarcrm_dev/blob/master/composer.json.
>
> I want to union SugarCRM with Laravel5.
> I added console support in SugarCRM, Monolog, generator _ide_helper.php
> https://github.com/butschster/sugarcrm_dev/blob/master/_ide_helper.php
> with phpDoc for all beans.
>
> Perhaps you might be interested.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sugarcrm/sugarcrm_dev/pull/233#issuecomment-172608235>
> .
>

butschster commented:
CE - Community Edition

2016-01-19 12:07 GMT+03:00 Jakub Paś <notifications@github.com>:

> Sound interesting. I will take a look. I made some improvements to Sugar
> myself. Perhaps I can contribute. One question - what the CE stnads for?
>
> On Mon, Jan 18, 2016 at 7:04 PM, butschster <notifications@github.com>
> wrote:
>
> > I started to improve SugarCRM 6.5 CE. And i moved at first standart php
> > libraries (phpmailer, tcpdf, zf, tcpdf, e.t.c) to composer
> > https://github.com/butschster/sugarcrm_dev/blob/master/composer.json.
> >
> > I want to union SugarCRM with Laravel5.
> > I added console support in SugarCRM, Monolog, generator _ide_helper.php
> > https://github.com/butschster/sugarcrm_dev/blob/master/_ide_helper.php
> > with phpDoc for all beans.
> >
> > Perhaps you might be interested.
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/sugarcrm/sugarcrm_dev/pull/233#issuecomment-172608235>
>
> > .
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sugarcrm/sugarcrm_dev/pull/233#issuecomment-172782110>
> .
>



-- 
Best regards,
*Pavel Buchnev*

*Mobile: +7* *926 *569 61 13
*CV/Resume:* www.bpy.me
*Facebook: *butschster <http://www.facebook.com/butschster>
*VK: *ButscH <http://vk.com/butsch>

chris001 commented:
@butschster @jakubpas Very cool. I'm requesting to pull your Composer and Laravel5 support into SuiteCRM also... http://github.com/salesagility/SuiteCRM

jakubpas commented:
Is this project active? I see a lot of pull requsts but seems they a neithe accepted nor rejected.
